CONTRIBUTING.md

```diff
@@ -870,6 +870,18 @@ benefits of this kind of change are very small, and in our experience it is not
 worth investing the substantial effort needed to review them. This especially
 includes changes suggested by tools.
 
+We normally immediately reject PRs which target platforms or system
+configurations that are not in the [official support
+matrix](https://www.elastic.co/support/matrix). We choose to support particular
+platforms with care because we must work to ensure that every Elasticsearch
+release works completely on every platform, and we must spend time
+investigating test failures and performance regressions there too. We cannot
+determine whether PRs which target unsupported platforms or configurations meet
+our quality standards, nor can we guarantee that the change they introduce will
+continue to work in future releases. It would be a very bad experience for our
+users if Elasticsearch suddenly stopped working on a particular platform after
```

Comment:
Maybe a tad less pointed...

```suggestion
- It would be a very bad experience for our users if Elasticsearch suddenly stopped working

+ We do not want Elasticsearch to suddenly stop working
```